src/main/java/com/sap/sgs/phosphor/fosstars/model/math/MathHelper.java

```diff
@@ -0,0 +1,122 @@
+package com.sap.sgs.phosphor.fosstars.model.math;
+
+import java.util.function.Function;
+import org.apache.commons.math3.analysis.function.Logistic;
+
+/**
+ * This is a helper class math.
+ */
+public class MathHelper {
+
+  /**
+   * Invert a value of a logistic function on a specified interval.
+   *
+   * @param func The logistic function to be inverted.
+   * @param a The beginning of the interval.
+   * @param b The end of the interval.
+   * @param value The value to be inverted.
+   * @param precision The precision for comparison of doubles.
+   * @return The argument of the function, which belongs to the specified interval [a, b],
+   *         which produces the closest value to the specified value.
+   * @throws IllegalArgumentException If the parameters are incorrect.
+   */
+  public static int invert(Logistic func, int a, int b, double value, double precision) {
+    return invert(func::value, a, b, value, precision);
+  }
+
+  /**
+   * Invert a value of a monotonic function on a specified interval.
+   * Note that the method expects a caller to pass a valid monotonic function.
+   * The method doesn't check whether the specified function is monotonic or not.
+   * If the function is not monotonic, the method may not produce a correct result.
+   *
+   * @param func The function to be inverted.
+   * @param a The beginning of the interval.
+   * @param b The end of the interval.
+   * @param value The value to be inverted.
+   * @param precision The precision for comparison of doubles.
+   * @return The argument of the function, which belongs to the specified interval [a, b],
+   *         which produces the closest value to the specified value.
+   * @throws IllegalArgumentException If the parameters are incorrect.
+   */
+  public static int invert(
+      Function<Integer, Double> func, int a, int b, double value, double precision) {
+
+    if (a > b) {
+      throw new IllegalArgumentException("a > b");
+    }
+    double funcA = func.apply(a);
```

Comment:
why don't you name it differently as valueA or something? because funcA is confusing if it's a function or a double value?

Reply:
`funcA` is supposed to mean "func(a)" which means "a value of the function in the point a". That's why it is `funcA`. For me it's not confusing. If you think it is confusing, please suggest how exactly you'd like to name it (with the meaning above). The shorter the better.

Reply:
`valueA` doesn't look better to me.

Reply:
It's of-course not confusing to you because you are the author. :) . Please see the below line:
if (equals(funcB, value, precision)) {

this looks like you are comparing a function to a value. 

Why don't you name it as 

functionValueA?

Reply:
As I said earlier, if you are writing the code for yourself, fine for me. If you are writing the code for others to contribute or other to understand also, consider changing it.

Reply:
I am not saying that the current name it not confusing. I am saying it is not confusing to me (for whatever reason). That's why I asked what you think would be a better name. Preferably a short name. How about `funcOfA` instead of `functionValueA`?

Reply:
Hmm, isn't it a Value after applying function to a Value A? or it's still a function?

Reply:
To my understanding, it's an inverted Value of A right?

Reply:
It is a value but I think `funcOfA` is clear enough. If it is still not, then I suggest `funcValueA`. Does it work for you?

Reply:
No funOfA is fine :) Thanks for the clarification.

Reply:
Taking into account your points, I will update it to `funcValueA` nevertheless.

Reply:
No, No, now for me funcOfA makes more sense.

Reply:
Sure, no problem. I have updated the variable name to `funcOfA`, `funcOfB` and so on. Please let me know if I missed anything.